src/components/BookForm.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
```

Comment:
Our `BookForm` is way too complicated:

1. #6 is directly happening here in this component and the root-cause of which is the "disArray" problem. And then because of that "disArray" problem we need to do a lot of complicated logic just to add a book.
2. Solving the "disArray" root problem will free us from ever needing anything from the redux store here aside from the `createBook` action we need to dispatch. So in that case we won't need `mapStateToProps` (also we don't need to declare `mapDispatchToProps` if you have read my previous comments about simplifying passing of actions to `connect`).

3. We don't need a factory in here. Had we simplified our logic starting from eliminating the "disArray" issue we would simply:

```
handleSubmit = (e) => {
  e.preventDefault();
  // assuming we transferred id creation into the reducer
  createBook({
   title,
   category
  });
 setTitle(' ');
 setCategory('Action');
}
```